packages/core/src/application.ts

```diff
@@ -280,141 +32,89 @@ export class Application extends Context {
-   * @param path URL path of the endpoint
-   * @param spec The OpenAPI spec describing the endpoint (operation)
-   * @param controller Controller constructor
-   * @param methodName The name of the controller method
+   * @param {Server} server The instantiated instance of your Server.
+   * @param {string=} name Optional override for key name.
+   * @returns {Binding}
+   * @memberof Application
    */
-  route(
-    verb: string,
-    path: string,
-    spec: OperationObject,
-    controller: ControllerClass,
-    methodName: string): Binding;
+  public server(server: Server, name?: string) : Binding {
+    const suffix = name || server.constructor.name;
+    const key = `servers.${suffix}`;
+    this._servers.push(key);
+    return this.bind(key).to(server);
+
+  }
 
   /**
-   * Register a new route.
+   * Bind an array of Server implementations to the Application's master
+   * context.
+   * Each server added in this way will automatically be named based on the
+   * class constructor name with the "servers." prefix.
    *
+   * If you wish to control the binding keys for particular server instances,
+   * use the app.server function instead.
    * ```ts
-   * function greet(name: string) {
-   *  return `hello ${name}`;
-   * }
-   * const route = new Route('get', '/', operationSpec, greet);
-   * app.route(route);
+   * app.servers([
+   *  new RestServer(app, options),
+   *  new GRPCServer(app, options),
+   * ]);
+   * // Creates a binding for "servers.RestServer" and a binding for
+   * // "servers.GRPCServer";
    * ```
    *
-   * @param route The route to add.
+   * @param {Array<Server>} servers An array of instantiated Server instances.
+   * @returns {Binding}
+   * @memberof Application
    */
-  route(route: RouteEntry): Binding;
-
-  route(
-    routeOrVerb: RouteEntry | string,
-    path?: string,
-    spec?: OperationObject,
-    controller?: ControllerClass,
-    methodName?: string,
-  ): Binding {
-    if (typeof routeOrVerb === 'object') {
-      const r = routeOrVerb;
-      return this.bind(`routes.${r.verb} ${r.path}`).to(r);
-    }
-
-    if (!path) {
-     throw new AssertionError({
-       message: 'path is required for a controller-based route',
-     });
-    }
-
-    if (!spec) {
-     throw new AssertionError({
-       message: 'spec is required for a controller-based route',
-     });
-    }
-
-    if (!controller) {
-     throw new AssertionError({
-       message: 'controller is required for a controller-based route',
-      });
-    }
-
-    if (!methodName) {
-      throw new AssertionError({
-        message: 'methodName is required for a controller-based route',
-      });
-    }
-
-    return this.route(new ControllerRoute(
-      routeOrVerb,
-      path,
-      spec,
-      controller,
-      methodName));
-  }
-
-  api(spec: OpenApiSpec): Binding {
-    return this.bind(CoreBindings.API_SPEC).to(spec);
+  public servers(servers: Server[]): Binding[] {
+    const self = this;
```

Comment:
Arrow functions (rocket functions) preserve `this`, there is no need to store it in `self`.

```ts
return servers.map(server => this.server(server));
```